.github/workflows/test.yml

```
@@ -3,14 +3,39 @@ on: pull_request
 env: { CLICOLOR_FORCE: 1 }
 
 jobs:
-  test:
+  cargo-fmt:
+    name: cargo-fmt@stable
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
       - name: Run formatting checks
         run: cargo fmt --check
+  cargo-clippy:
+    name: cargo-clippy@${{ matrix.version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: [1.63, stable, beta]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install Rust ${{ matrix.version }}
+        uses: dtolnay/rust-toolchain@${{ matrix.version }}
       - name: Run clippy checks
         run: cargo clippy --all-features -- -D warnings
+  cargo-test:
+    name: cargo-test@${{ matrix.version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: [1.63, stable, beta]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install Rust ${{ matrix.version }}
+        uses: dtolnay/rust-toolchain@${{ matrix.version }}
```

Pretty sure I tried this myself in the past and Github Actions didn't accept variables to specify the version.
```suggestion
        uses: dtolnay/rust-toolchain@master
        with:
            toolchain: ${{ matrix.version }}
```

---

That's fine with me, especially since addition components will have to be specified for some of these anyway.